lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/GeneratingSuggester.java

```diff
@@ -89,7 +89,8 @@
       }
 
       String lower = dictionary.toLowerCase(root);
-      int sc = ngram(3, word, lower, options) + commonPrefix(word, root);
+      int sc =
+          automaton.ngramScore(lower) - longerWorsePenalty(word, lower) + commonPrefix(word, root);
```

Comment:
`ngram` has been split into `ngramScore` and penalties based on previously passed flags. The former is replaced with the automaton.